python/cudf/cudf/core/multiindex.py

```diff
@@ -179,7 +185,7 @@ def copy(self, deep=True):
         if self._levels is not None:
             mi._levels = [s.copy(deep) for s in self._levels]
         if self._codes is not None:
-            mi._codes = self._codes.copy(deep)
+            mi._codes = self._codes.copy()
```

Comment:
This is now always deep copying. We should shallow copy if `deep=False`:
```suggestion
            mi._codes = self._codes.copy(deep)
```